src/Views/FirmwareView.vala

```diff
@@ -92,67 +96,62 @@ public class About.FirmwareView : Gtk.Stack {
 
         num_updates = 0;
 
-        foreach (var device in yield fwupd.get_devices ()) {
-            add_device (device);
-        }
+        FirmwareClient.get_devices.begin (client, (obj, res) => {
+            try {
+                var devices = FirmwareClient.get_devices.end (res);
+                for (int i = 0; i < devices.length; i++) {
+                    add_device (client, devices[i]);
+                }
+
+                update_list.show_all ();
+            } catch (Error e) {
+                // TODO: Use placeholder
+                critical (e.message);
+            }
+        });
```

Comment:
If you want to await the result of an async method inside an async method, you can use `yield` instead of `begin`/`end`. 

The original code that was removed here was:
```
 foreach (var device in yield fwupd.get_devices ()) {
    add_device (device);
}
```

That can pretty much be put back using your new method.